decidim-bulletin_board-js/src/voter/voter.js

```diff
@@ -32,23 +32,43 @@ export class Voter {
     return this.wrapper.encrypt(data);
   }
 
-  async verifyVote(voteHash) {
-    const { id: electionUniqueId } = this.electionContext;
-
-    return new Promise((resolve) => {
+  /**
+   * Confirms if a vote was processed
+   *
+   * @param {Object} messageId - An object that includes the following options.
+   *  - {String} messageId - The messageId
+   * @returns {Promise<Object>} - Returns the PendingMessage
+   */
+  waitForPendingMessageToBeProcessed(messageId) {
+    return new Promise((resolve, reject) => {
       const intervalId = setInterval(() => {
         this.bulletinBoardClient
-          .getLogEntry({
-            electionUniqueId,
-            contentHash: voteHash,
+          .getPendingMessageByMessageId({
+            messageId,
           })
-          .then((logEntry) => {
-            if (logEntry) {
+          .then((pendingMessage) => {
+            if (pendingMessage.status !== "enqueued") {
               clearInterval(intervalId);
-              resolve();
+              resolve(pendingMessage);
             }
           });
       }, this.options.bulletinBoardWaitTime);
     });
   }
+
+  /**
+   * Verifies a vote
+   *
+   * @param {Object} voteHash - An object that includes the following options.
+   *  - {String} voteHash - the voteHash of a vote
```

Comment:
You are just giving it the `voteHash` as a `String`, not an object 👍 . Also I recommend renaming it to `contentHash` so you can save some keystrokes.

Reply:
Great suggestion! Thanks a lot!